recipe/activate-go-core.bat

```diff
@@ -1,6 +1,2 @@
 @set "CONDA_GOROOT_BACKUP=%GOROOT%"
 @set "GOROOT=%CONDA_PREFIX%\go"
-
-if "%CONDA_BUILD%"==1 (
-  set GOPATH=%SRC_DIR%
```

Comment:
Curious why this is being dropped?

Reply:
For consistency with the .sh activate scripts I moved it to activate-go-platform.bat.